app/services/govspeak_document/in_app_options.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+class GovspeakDocument::InAppOptions < GovspeakDocument::Options
+  include Rails.application.routes.url_helpers
+
+  def to_h
+    super.to_h.merge(images: in_app_images)
+  end
+
+private
+
+  def in_app_images
+    edition.revision.image_revisions.map do |image_revision|
+      image_attributes(image_revision).merge(
+        # This is the same as url_for(image_revision.crop_variant) in the view, or
+        # image_revision.crop_variant.service_url, except that these don't work outside
+        # of a request context which normally includes the request host.
```

Comment:
This seems quite a big comment to have in a merge, it'd probably be easier to follow if we assigned a variable to this with the comment and then injected it in. 

For what it's worth though I don't think this comment is that necessary as I don't really know that our usage of `service_url`  is advised behaviour by activestorage and I don't think we actually use it in the app already (based on a quick grep)

Reply:
I think it's necessary, as it's not something that's covered in the documentation - I've updated the comment to make this clearer.